Web/WebService_Session_2/src/com/employee/service/EmployeeService.java

```diff
@@ -0,0 +1,255 @@
+						+ e.getMessage() + "]");
+				throw new EmployeeSystemException(
+						"Coult not commit the transaction, [" + e.getMessage()
+								+ "]", e);
+			}
+		} catch (Exception e) {
+			try {
+				System.out.println("could not get employee by name,["
+						+ e.getMessage() + "]");
+				connection.rollback();
+				System.out.println("Transaction roll back");
+				throw new EmployeeSystemException(
+						"could not get employee by name,Transaction roll back,["
+								+ e.getMessage() + "]", e);
+			} catch (SQLException e1) {
+				System.out.println("Error in Transaction roll back,["
+						+ e1.getMessage() + "]");
+				throw new EmployeeSystemException(
+						"Error in  Transaction roll back,[" + e1.getMessage()
+								+ "]", e);
+			}
+		} finally {
+			ConnectionFactory.closeConnection(connection);
+		}
+		return employees;
+	}
+
+	public List<Employee> retrieveAll() throws EmployeeSystemException {
+		Connection connection = null;
+		try {
+			connection = ConnectionFactory.getConnection();
+
+		} catch (EmployeeSystemException e) {
+			System.out.println("Coult not create connection with database ["
+					+ e.getMessage() + "]");
+			throw new EmployeeSystemException(
+					"Coult not create connection with database ["
+							+ e.getMessage() + "]", e);
+		}
+		EmployeeFacade employeeFacade = new EmployeeFacade();
+		List<Employee> employees;
+		try {
+			employees = employeeFacade.retrieveAll(connection);
+
+			try {
+				connection.commit();
+			} catch (SQLException e) {
+				System.out.println("Coult not commit the transaction, ["
+						+ e.getMessage() + "]");
+				throw new EmployeeSystemException(
+						"Coult not commit the transaction, [" + e.getMessage()
+								+ "]", e);
+			}
+		} catch (Exception e) {
+			try {
+				System.out.println("could not get all employees,["
+						+ e.getMessage() + "]");
+				connection.rollback();
+				System.out.println("Transaction roll back");
+				throw new EmployeeSystemException(
+						"could not get all employees,Transaction roll back,["
+								+ e.getMessage() + "]", e);
+			} catch (SQLException e1) {
+				System.out.println("Error in Transaction roll back,["
+						+ e1.getMessage() + "]");
+				throw new EmployeeSystemException(
+						"Error in  Transaction roll back,[" + e1.getMessage()
+								+ "]", e);
+			}
+		} finally {
+			ConnectionFactory.closeConnection(connection);
+		}
+		return employees;
+	}
+
+	public void deleteById(int id) throws EmployeeSystemException {
+		Connection connection = null;
+		try {
+			connection = ConnectionFactory.getConnection();
+
+		} catch (EmployeeSystemException e) {
+			System.out.println("Coult not create connection with database ["
+					+ e.getMessage() + "]");
+			throw new EmployeeSystemException(
+					"Coult not create connection with database ["
+							+ e.getMessage() + "]", e);
+		}
+		EmployeeFacade employeeFacade = new EmployeeFacade();
+
+		try {
+			employeeFacade.deleteById(id, connection);
```

Comment:
Why so u need to pass 'connection' as a parameter to the method can't this method create a connection by itself?